tests/distributions/test_hmm.py

```diff
@@ -95,15 +95,22 @@ def test_discrete_hmm_shape(ok, init_shape, trans_shape, obs_shape, event_shape,
     obs_dist = dist.Bernoulli(logits=obs_logits).to_event(len(event_shape))
     data = obs_dist.sample()[(slice(None),) * len(obs_shape) + (0,)]
 
-    if ok:
-        d = dist.DiscreteHMM(init_logits, trans_logits, obs_dist)
-        actual = d.log_prob(data)
-        expected_shape = broadcast_shape(init_shape, trans_shape[:-1], obs_shape[:-1])
-        assert actual.shape == expected_shape
-        check_expand(d, data)
-    else:
+    if not ok:
         with pytest.raises(ValueError):
             dist.DiscreteHMM(init_logits, trans_logits, obs_dist)
+        return
+
+    d = dist.DiscreteHMM(init_logits, trans_logits, obs_dist)
+
+    actual = d.log_prob(data)
+    expected_shape = broadcast_shape(init_shape, trans_shape[:-1], obs_shape[:-1])
+    assert actual.shape == expected_shape
+    check_expand(d, data)
+
+    final = d.filter(data)
+    assert isinstance(final, dist.Categorical)
+    assert final.batch_shape == d.batch_shape
```

Comment:
add assert for `support` to make sure hidden_dim is right?

Reply:
done.